File: PADTiltViewControllerExample/PADTiltViewControllerExample/PADFeedDataSource.m
//
//  PADFeedDataSource.m
//  PADTiltViewControllerExample
//
//  Created by Pavel Dusatko on 4/30/14.
//  Copyright (c) 2014 Pavel Dusatko. All rights reserved.
//

#import "PADFeedDataSource.h"

static NSString *CellIdentifier = @"PADFeedCell";

@interface PADFeedDataSource ()

@property (nonatomic, strong) NSArray *array;

@end

@implementation PADFeedDataSource

#pragma mark - Accessors

- (NSArray *)array {
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        NSString *path = [[NSBundle mainBundle] pathForResource:@"PADTiltViewControllerExample-Configuration" ofType:@"plist"];
        NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile:path];
        _array = dict[@"feed"];
    });
    return _array;
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.array.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"PADFeedCell" forIndexPath:indexPath];
    NSDictionary *dict = self.array[indexPath.row];
    
    UILabel *textLabel = (UILabel *)[cell viewWithTag:100];
    UITextField *detailTextField = (UITextField *)[cell viewWithTag:101];
    UIImageView *imageView = (UIImageView *)[cell viewWithTag:102];
    
    textLabel.text = dict[@"text"];
    detailTextField.text = dict[@"detailText"];
    imageView.image = [UIImage imageNamed:dict[@"imageName"]];
    
    return cell;
}

#pragma mark - UITableViewDelegate

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    NSDictionary *dict = self.array[indexPath.row];
    NSString *heightKeyPath = (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPhone) ? @"height.phone" : @"height.pad";
    return [[dict valueForKeyPath:heightKeyPath] floatValue];
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
}

@end

File: Swift-version/Inpiration/PADFeedDataSource.swift
//
//  PADFeedDataSource.swift
//  Inpiration
//
//  Created by Alexander Hüllmandel on 11/1/14.
//  Copyright (c) 2014 Alexander Hüllmandel. All rights reserved.
//

import Foundation
import UIKit


class PADFeedDataSource: NSObject, UITableViewDelegate, UITableViewDataSource {
    let CellIdentifier = "PADFeedCell"
    
    private lazy var array : Array<Dictionary<String, AnyObject>> = {
        let path = NSBundle.mainBundle().pathForResource("PADTiltViewControllerExample-Configuration", ofType: "plist")
        let dict = NSDictionary(contentsOfFile: path!)
        return dict!["feed"] as Array<Dictionary<String, AnyObject>>
    }()
    
    //MARK: - UITableViewDataSource
    
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return self.array.count
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        let sectionInfo = array[section] as Dictionary
        return sectionInfo.count
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCellWithIdentifier(CellIdentifier, forIndexPath: indexPath) as UITableViewCell
        let dict = array[indexPath.row]
        
        let textLabel = cell.viewWithTag(100) as? UILabel
        let detailTextField = cell.viewWithTag(101) as? UITextField
        let imageView = cell.viewWithTag(102) as? UIImageView
        
        textLabel?.text = dict["text"] as? String
        detailTextField?.text = dict["detailText"] as? String
        imageView?.image = UIImage(named: dict["imageName"] as String)
        
        return cell;
    }
    
//MARK: - UITableViewDelegate
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
//        let dict = self.array[indexPath.row]
//        let heightKey = "heigth"
//        let heightDict = dict[heightKey] as Dictionary<String,
//        let deviceKey = (UIDevice.currentDevice().userInterfaceIdiom == .Phone) ? "phone" : "pad"
        return 88.0
    }

    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        tableView.deselectRowAtIndexPath(indexPath, animated: true)
    }
}